monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -0,0 +1,370 @@
+        self.output_path = None
+        self.name = None
+        self.best_metric = None
+
+    def load_templates(self, config_files, metadata_file=None, parser_args=None):
+        """
+        Read a list of template configuration files
+
+        Args:
+            config_file: bundle config files.
+            metadata_file: metadata overriding file
+            parser_args: argument to parse
+
+        """
+        parser_args = parser_args or {}
+        self.cfg.read_config(config_files, **parser_args)
+        if metadata_file is not None:
+            self.cfg.read_meta(metadata_file)
+
+    def set_data_stats(self, data_stats_files: str):  # type: ignore
+        """
+        Set the data anlysis report (generated by DataAnalyzer).
+
+        Args:
+            data_stats_files: path to the datastats yaml file
+        """
+        self.data_stats_files = data_stats_files
+
+    def set_data_source(self, data_list_file: str):
+        """
+        Set the data source configuration file
+
+        Args:
+            data_list_file: path to a configuration file (yaml) that contains datalist, dataroot, and other params.
+        """
+        self.data_list_file = data_list_file
+
+    def fill_template_config(self, data_stats_filename, **kwargs):
+        """
+        The configuration files defined when constructing this Algo instance might not have a complete training
+        and validation pipelines. Some configuration components and hyperparameters of the pipelines depend on the
+        training data and other factors. This API is provided to allow the creation of fully functioning config files.
+
+        Args:
+            data_stats_filename: filename of the data stats report (generated by DataAnalyzer)
+        """
+        pass
+
+    def export_to_disk(self, output_path: str, algo_name: str, **kwargs):
+        """
+        Fill the configuration templates, write the bundle (configs + scripts) to folder `output_path/algo_name`.
+
+        Args:
+            output_path: Path to export the 'scripts' and 'configs' directories.
+            algo_name: the identifier of the algorithm (usually contains the name and extra info like fold ID).
+        """
+        self.fill_template_config(self.data_stats_files, **kwargs)
+        write_path = os.path.join(output_path, algo_name)
+        self.cfg["bundle_root"] = write_path
+        os.makedirs(write_path, exist_ok=True)
+        # handling scripts files
+        output_scripts_path = os.path.join(write_path, "scripts")
+        if os.path.exists(output_scripts_path):
+            shutil.rmtree(output_scripts_path)
+        if self.scripts_path is not None and os.path.exists(self.scripts_path):
+            shutil.copytree(self.scripts_path, output_scripts_path)
+        # handling config files
+        output_config_path = os.path.join(write_path, "configs")
+        if os.path.exists(output_config_path):
+            shutil.rmtree(output_config_path)
```

Comment:
this `rmtree` might remove existing experiments and results...kind of dangerous here but I don't have a better idea for now.

Reply:
I added a few line to copy the existing experiments and results to a backup (.bak) and a warning message.